File: aquantica-api/src/Aquantica.Core/Entities/IrrigationSection.cs
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace Aquantica.Core.Entities;

[Table("IrrigationSections")]
public class IrrigationSection : BaseEntity
{
    public int Number { get; set; }
    public string? Name { get; set; }
    public int? ParentId { get; set; }
    public bool IsEnabled { get; set; }
    public int? LocationId { get; set; }
    public virtual Location? Location { get; set; }
    public virtual IrrigationSection? ParentSection { get; set; }
    public virtual ICollection<IrrigationEvent>? IrrigationEvents { get; set; }
    public int? SectionRulesetId { get; set; }
    public virtual IrrigationRuleset? IrrigationRuleset { get; set; }
    public int? IrrigationSectionTypeId { get; set; }
    public virtual IrrigationSectionType? IrrigationSectionType { get; set; }
    
    public virtual ICollection<BackgroundJob>? BackgroundJobs { get; set; }
    
}

File: aquantica-web/src/app/@core/models/irrigation-section.ts
export interface IrrigationSection {
  id: number;
  number: number;
  name?: string | null; // Assuming `string?` corresponds to an optional string in TypeScript
  parentId?: number | null;
  isEnabled: boolean;
  deviceUri: string;
  sectionRulesetId?: number | null;
  parentNumber?: number | null;
  locationId?: number | null;
}
